src/graphnet/data/datamodule.py

```diff
@@ -0,0 +1,429 @@
+"""Base `Dataloader` class(es) used in `graphnet`."""
+from typing import Dict, Any, Optional, List, Tuple, Union
+import pytorch_lightning as pl
+from torch.utils.data import DataLoader
+from copy import deepcopy
+from sklearn.model_selection import train_test_split
+import pandas as pd
+
+from graphnet.data.dataset import (
+    Dataset,
+    EnsembleDataset,
+    SQLiteDataset,
+    ParquetDataset,
+)
+from graphnet.utilities.logging import Logger
+
+
+class GraphNeTDataModule(pl.LightningDataModule, Logger):
+    """General Class for DataLoader Construction."""
+
+    def __init__(
+        self,
+        dataset_reference: Union[SQLiteDataset, ParquetDataset, Dataset],
+        dataset_args: Dict[str, Any],
+        selection: Optional[Union[List[int], List[List[int]]]] = None,
+        test_selection: Optional[Union[List[int], List[List[int]]]] = None,
+        train_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+        validation_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+        test_dataloader_kwargs: Optional[Dict[str, Any]] = None,
+        train_val_split: Optional[List[float]] = [0.9, 0.10],
+        split_seed: int = 42,
+    ) -> None:
+        """Create dataloaders from dataset.
+
+        Args:
+            dataset_reference: A non-instantiated reference to the dataset class.
+            dataset_args: Arguments to instantiate graphnet.data.dataset.Dataset with.
+            selection: (Optional) a list of event id's used for training and validation, Default None.
+            test_selection: (Optional) a list of event id's used for testing, Default None.
```

Comment:
A few of these lines are longer than 79 characters. Could you break them up onto multiple lines, please? VSCode has the ability to displace a vertical ruler in the editor  that indicates character length. See [here](https://dev.to/brad_beggs/vs-code-vertical-rulers-for-prettier-code-3gp3). 

E.g: 

```python
"""
dataset_reference: A non-instantiated reference to the dataset class.
dataset_args: Arguments to instantiate 
            graphnet.data.dataset.Dataset with.
selection: (Optional) a list of event id's used for training
           and validation, Default None.
"""
```